packages/mui-base/src/Slider/Root/SliderRoot.tsx

```diff
@@ -0,0 +1,206 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { sliderStyleHookMapping } from './styleHooks';
+import { useSliderRoot } from './useSliderRoot';
+import { SliderProvider } from './SliderProvider';
+import { SliderRootProps, SliderRootOwnerState } from './SliderRoot.types';
+
+const SliderRoot = React.forwardRef(function SliderRoot(
+  props: SliderRootProps,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const {
+    'aria-labelledby': ariaLabelledby,
+    className,
+    defaultValue,
+    direction = 'ltr',
+    disabled = false,
+    largeStep,
+    render,
+    minStepsBetweenValues,
+    onValueChange,
+    onValueCommitted,
+    orientation = 'horizontal',
+    value,
+    ...otherProps
+  } = props;
+
+  const { getRootProps, ...slider } = useSliderRoot({
+    'aria-labelledby': ariaLabelledby,
+    defaultValue,
+    disabled,
+    direction,
+    largeStep,
+    minStepsBetweenValues,
+    onValueChange,
+    onValueCommitted,
+    orientation,
+    rootRef: forwardedRef,
+    value,
+    ...otherProps,
+  });
+
+  const ownerState: SliderRootOwnerState = React.useMemo(
+    () => ({
+      activeThumbIndex: slider.active,
+      direction,
+      disabled,
+      dragging: slider.dragging,
+      orientation,
+      max: slider.max,
+      min: slider.min,
+      minStepsBetweenValues: slider.minStepsBetweenValues,
+      step: slider.step,
+      values: slider.values,
+    }),
+    [
+      direction,
+      disabled,
+      orientation,
+      slider.active,
+      slider.dragging,
+      slider.max,
+      slider.min,
+      slider.minStepsBetweenValues,
+      slider.step,
+      slider.values,
+    ],
+  );
+
+  const contextValue = React.useMemo(
+    () => ({
+      ...slider,
+      ownerState,
+    }),
+    [slider, ownerState],
+  );
+
+  const { renderElement } = useComponentRenderer({
+    propGetter: getRootProps,
+    render: render ?? 'div',
+    ownerState,
+    className,
+    ref: forwardedRef,
```

Comment:
```suggestion
```

The `forwardedRef` will overwrite the merged ref returned from the hook.

Reply:
Fixed!